File: workplace-be/src/main/java/com/emetaplus/workplace/profile/model/FullProfile.java
package com.emetaplus.workplace.profile.model;

import com.emetaplus.workplace.builder.banner.model.Banner;
import com.emetaplus.workplace.builder.contact.model.ProfileContact;
import com.emetaplus.workplace.builder.decks.model.ProfileDeck;
import com.emetaplus.workplace.builder.schedule.model.Schedule;
import com.emetaplus.workplace.builder.shortinfo.model.ShortInfo;
import lombok.Data;

import java.util.UUID;

@Data
public class FullProfile {
    private UUID profileId;
    private String profileURL;
    private Banner banner;
    private ShortInfo shortInfo;
    private Schedule schedule;
    private ProfileDeck profileDeck;
    private ProfileContact profileContact;
}

File: workplace-fe/src/app/pages/full-profile/full-profile.ts
import {Banner} from "../profile/builder/banner/banner";
import {ShortInfo} from "../profile/builder/short-info/short-info";
import {Schedule} from "../profile/builder/schedule/schedule";
import {ProfileDeck} from "../profile/builder/my-decks/profile-deck";
import {ProfileContact} from "../profile/builder/contacts/profile-contact";

export class FullProfile {
  profileId: string | undefined;
  profileURL: string | undefined;
  banner: Banner = new Banner();
  shortInfo: ShortInfo = new ShortInfo();
  schedule: Schedule = new Schedule();
  profileDeck: ProfileDeck = new ProfileDeck();
  profileContact: ProfileContact = new ProfileContact();
}
